x/staking/types/params.go

```diff
@@ -108,11 +109,70 @@ func UnmarshalParams(cdc *codec.Codec, value []byte) (params Params, err error)
 
 // validate a set of params
 func (p Params) Validate() error {
-	if p.BondDenom == "" {
-		return fmt.Errorf("staking parameter BondDenom can't be an empty string")
+	if err := validateUnbondingTime(p.UnbondingTime); err != nil {
+		return err
 	}
-	if p.MaxValidators == 0 {
-		return fmt.Errorf("staking parameter MaxValidators must be a positive integer")
+	if err := validateMaxValidators(p.MaxValidators); err != nil {
+		return err
 	}
+	if err := validateMaxEntries(p.MaxEntries); err != nil {
+		return err
+	}
+	if err := validateBondDenom(p.BondDenom); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateUnbondingTime(i interface{}) error {
+	v, ok := i.(time.Duration)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if v <= 0 {
+		return fmt.Errorf("unbonding time must be positive: %d", v)
+	}
+
+	return nil
+}
+
+func validateMaxValidators(i interface{}) error {
+	v, ok := i.(uint16)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if v == 0 {
+		return fmt.Errorf("max validators must be positive: %d", v)
+	}
+
+	return nil
+}
+
+func validateMaxEntries(i interface{}) error {
+	v, ok := i.(uint16)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if v == 0 {
+		return fmt.Errorf("max entries must be positive: %d", v)
+	}
+
+	return nil
+}
+
+func validateBondDenom(i interface{}) error {
+	v, ok := i.(string)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if strings.TrimSpace(v) == "" {
```

Comment:
Should denomination be alphabetic as well (no space as well)?

Reply:
We can call denom validation, yes :)